### ISSUE TITLE
Filter out halo from bright spots

### ISSUE THREAD

**Issue by PetteriAimonen** [PetteriAimonen/focus-stack#8]
When the object has sharp bright spots, they get blurred on other layers of the stack.
This causes false edges that get included in final image.
It would be nice if this could be filtered out somehow.

**Comment by pjrose** [comment]
Ran this 12 times in a row, no more lockup. Nice!

**Comment by soad89** [comment]
Hello. 
Can you please have a look on attached stack? Is that the issue you are referring to? The result img is inside too.
Thanks
[stack.zip](https://github.com/PetteriAimonen/focus-stack/files/11270941/stack.zip)

**Comment by PetteriAimonen** [comment]
@soad89  Yes, it is. This is a quite typical case, the background is never in focus and the bright areas spread on top of it and get mistaken as details.

**Comment by pjkkerr** [comment]
My first focus-stack focus stack was quite successful, apart from tiny highlights blowing up into ghastly artifacts like this:
<img width="1922" height="1640" alt="Image" src="https://github.com/user-attachments/assets/e84ab893-e711-4d55-a626-964bf8d14509" />
I got rid of the worst of them using the clone tool in Gimp, but there are still many more smaller ones.
My 2c worth on how to solve this particular case would be as follows. If we have coordinates x,y,z where z represents the stack, presumably we have an "in focus" metric F as a function of z, and for each x,y pair we find the z that maximizes F. Could we specify a brightness threshold so that we maximize F subject to final pixel brightness<threshold. If none qualify we've got a genuine blown highlight, so use the brightest pixel in the stack. I wish my software skills were up to trying this myself, but when I look at the code... nah.
Having posted this comment and then looked hard at the left and right images I've noticed the stacked image is much softer, which wouldn't actually be a problem at the size I might want to print this image.

**Comment by PetteriAimonen** [comment]
@pjkkerr Yeah, that's probably the basic approach, the difficulty is in the details and in how to make it interact with the wavelet processing.
